### ISSUE TITLE
DPO/PPO/etc should default to HF chat template

### ISSUE THREAD

**Issue by hamishivi** [allenai/open-instruct#458]
Currently, our codebase is somewhat hard-coded when it comes to chat templates. We should mainly assume the SFT stage correctly sets the chat template in the tokenizer (which it currently does) and just load from that in all post-SFT training stages. We can probably have a few alternate options if it makes sense.

**Comment by vwxyzjn** [comment]
Yeah this is a good point. We should put this logic all in some sort of tokenization module that can be re-used for SFT / DPO / RL

**Comment by hamishivi** [comment]
This got fixed in #513
